moesifwsgi/middleware.py

```diff
@@ -76,8 +74,8 @@ def __init__(self, app, settings):
         self.config = self.app_config.get_config(self.api_client, self.DEBUG)
         self.sampling_percentage = 100
         self.last_updated_time = datetime.utcnow()
-        self.moesif_events_queue = queue.Queue()
-        self.BATCH_SIZE = self.settings.get('BATCH_SIZE', 25)
+        self.moesif_events_queue = queue.Queue(self.settings.get('EVENT_QUEUE_SIZE', 100000))
```

Comment:
100K is a large for most of the standard size VMs . We can keep it 1K and request transpose / other customer with large VMs to raise the value.

Reply:
@praves77 @bakennedy , was concerned 1k was too small and risks dropping data. Assuming 1KB event size (which is our typical event size), 10k would be 10MB which provides plenty of buffer. Even if the customer had the extreme of 100KB events, then we're still only 1GB. Maybe 10k is happy middle?

Reply:
Got it. 10K is probably better because at the worst case we would have 2x (of 10K events) in memory:
1. one for ingestion queue
2. 2nd one (a snapshot of full queue) ready for upload.
Only in this scenario events will be dropped as ingestion queue is also full.

Either one is fine - we need test results to understand behavior.

Reply:
There are two queues, and they both have a Max size. Python objects are passed by reference, so the only thing being doubled is a list of pointers to the objects rather than their original data so that's good.

 The max size on the queue of batches is less likely to ever be hit since there are 1% as many batches as events. The max size on the event queue will be hit when there are enough events that that queue fills up while they are being batched and put in the batch queue. This would happen for example if the job were fully saturated with CPU activity and The batch thread is unable to drain the input queue. 

The batch queue could be given a proportionately smaller maximum size if we wanted to ensure that it filling up does not result in running out of memory

Reply:
@praves77 @dgilling I added the default queue numbers discussed above, and ensure that the batch queue holds the same number of total events as the event queue to prevent unexpected high memory usage